tests/test_client.py

```diff
@@ -0,0 +1,197 @@
+import unittest
+import boto3
+import docker
+import botor
+import pytest
+
+
+def create_botwo_client(minio1, minio2):
+    profiles = {
+        "test_upload": {
+            "source_bucket_pattern": "bucket-a",
+            "source_key_pattern": "a/*",
+            "mapped_bucket_name": "test-upload",
+            "mapped_prefix": "main/"
+        },
+        "test_copy": {
+            "source_bucket_pattern": "source",
+            "mapped_bucket_name": "test-copy",
+            "mapped_prefix": "main/"
+        },
+        "test_copy2": {
+            "source_bucket_pattern": "test-copy",
+        },
+        "test_delete_objects": {
+            "source_bucket_pattern": "delete-objects*",
+            "mapped_bucket_name": "test-delete-objects",
+            "mapped_prefix": "test/"
+        },
+        "test_paginator": {
+            "source_bucket_pattern": "pagi",
+            "mapped_bucket_name": "test-paginator",
+            "mapped_prefix": "dev/"
+        },
+        "test_list_objects": {
+            "source_bucket_pattern": "list",
+            "mapped_bucket_name": "test-list-objects",
+            "mapped_prefix": "dev/"
+        }
+
+    }
+    client_mapping = {"test_upload": minio1, "test_copy": minio1, "test_copy2": minio1,
+                      "test_delete_objects": minio2, "test_paginator": minio2, "test_list_objects": minio2,
+                      "default": minio2}
```

Comment:
AFAICT this means that we test most actions only for being routed to one bucket.  So our testing looks like at most half of what it should be.